src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs

```diff
@@ -236,6 +243,34 @@ public SignatureTrustAndValidityVerificationProvider()
                 timestampSummary.Issues.Concat(status.Issues));
         }
 
+        private bool IsUntrustedRootAllowed(Signature signature)
+        {
+            if (_allowUntrustedRootList != null)
+            {
+                foreach (var allowUntrustedRoot in _allowUntrustedRootList)
+                {
+                    try
+                    {
+                        var signingCertificateFingerprint = signature.GetSigningCertificateFingerprint(allowUntrustedRoot.Value);
+
+                        if (string.Equals(allowUntrustedRoot.Key, signingCertificateFingerprint, StringComparison.Ordinal))
+                        {
+                            return true;
+                        }
+                    }
+                    // if the exception is InvalidOperationException it means the signature is invalid
```

Comment:
When would this be thrown?

Reply:
`GetSigningCertificateFingerprint(…)` will throw `InvalidOperationException` if the signature does not have a signing certificate. Therefore, the whole signature is bad and this validation won't be possible, so that's why I just return false in that case.

Reply:
Makes sense.